comp/api/api/apiimpl/observability/telemetry_test.go

```diff
@@ -146,3 +147,11 @@ func TestTelemetryHandlerShort(t *testing.T) {
 	duration := time.Since(start).Milliseconds()
 	require.LessOrEqual(t, duration, 100*time.Millisecond)
 }
+
+func TestTelemetryMiddlewareTwice(t *testing.T) {
```

Comment:
We have this test to ensure we do not panic on registering the same metric twice. Should we add a comment or change the test name to explain why we have it?

Reply:
Sure